django/db/models/query_utils.py

```diff
@@ -169,6 +173,78 @@ def _check_parent_chain(self, instance, name):
         return None
 
 
+class RegisterLookupMixin(object):
+    def _find_lookup(self, lookup_name, class_attr):
+        """
+        Finds lookups or transforms registered on this, or any parent, class.
+        `class_attr` can be either `class_lookups` or `class_transforms`.
+        """
+        try:
+            return getattr(self, class_attr)[lookup_name]
+        except KeyError:
+            # To allow for inheritance, check parent class'
+            for parent in inspect.getmro(self.__class__):
+                parent_lookups = getattr(parent, class_attr, None)
+                if not parent_lookups:
+                    continue
+                if lookup_name in parent_lookups:
+                    return parent_lookups[lookup_name]
+        except AttributeError:
+            # Nothing has been registered
+            pass
+        return None
+
+    def get_lookup(self, lookup_name):
+        found = self._find_lookup(lookup_name, 'class_lookups')
+        if found is None and hasattr(self, 'output_field'):
+            return self.output_field.get_lookup(lookup_name)
+        return found
+
+    def get_transform(self, lookup_name):
+        found = self._find_lookup(lookup_name, 'class_transforms')
+        if found is None and hasattr(self, 'output_field'):
+            return self.output_field.get_transform(lookup_name)
+        return found
+
+    @classmethod
+    def register_lookup(cls, lookup, lookup_name=None):
+        from django.db.models.lookups import Transform
```

Comment:
For clarity here, shouldn't we use `Func` rather than `Transform`, since they are equivalent and the latter is a back-compat-only name? It seems like using `Transform` might suggest to someone reading this code that there's something distinct about `Transform` as opposed to `Func`.